### PR TITLE
Replace SimpleCamera with PinholeCameraCal3_S2

### PR DESCRIPTION
**Pull request by varunagrawal**
Since `SimpleCamera` is deprecated, this PR makes `PinholeCamera<Cal3_S2>` the default in all the code, scripts, and examples.

This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/258)

### CODE REVIEW

File: cython/gtsam/examples/ImuFactorExample2.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def IMU_example():
     up = gtsam.Point3(0, 0, 1)
     target = gtsam.Point3(0, 0, 0)
     position = gtsam.Point3(radius, 0, 0)
-    camera = gtsam.SimpleCamera.Lookat(position, target, up, gtsam.Cal3_S2())
+    camera = gtsam.PinholeCameraCal3_S2.Lookat(position, target, up, gtsam.Cal3_S2())
```
Comment (dellaert) [inline cython/gtsam/examples/ImuFactorExample2.py]: import it?

File: cython/gtsam/utils/visual_data_generator.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def generate_data(options):
     for i in range(options.nrCameras):
         theta = i * 2 * pi / options.nrCameras
         t = gtsam.Point3(r * cos(theta), r * sin(theta), height)
-        truth.cameras[i] = gtsam.SimpleCamera.Lookat(t,
+        truth.cameras[i] = gtsam.PinholeCameraCal3_S2.Lookat(t,
```
Comment (dellaert) [inline cython/gtsam/utils/visual_data_generator.py]: import

File: examples/ImuFactorExample2.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@ int main(int argc, char* argv[]) {
   double radius = 30;
   const Point3 up(0, 0, 1), target(0, 0, 0);
   const Point3 position(radius, 0, 0);
-  const SimpleCamera camera = SimpleCamera::Lookat(position, target, up);
+  const PinholeCamera<Cal3_S2> camera = PinholeCamera<Cal3_S2>::Lookat(position, target, up);
```
Comment (dellaert) [inline examples/ImuFactorExample2.cpp]: auto

File: gtsam/sam/tests/testRangeFactor.cpp [hunk 4]
```diff
@@ -353,11 +354,11 @@ TEST(RangeFactor, Point3) {
 }
 
 /* ************************************************************************* */
-// Do tests with SimpleCamera
+// Do tests with PinholeCamera<Cal3_S2>
 TEST( RangeFactor, Camera) {
-  RangeFactor<SimpleCamera,Point3> factor1(poseKey, pointKey, measurement, model);
-  RangeFactor<SimpleCamera,Pose3> factor2(poseKey, pointKey, measurement, model);
-  RangeFactor<SimpleCamera,SimpleCamera> factor3(poseKey, pointKey, measurement, model);
+  RangeFactor<PinholeCamera<Cal3_S2>,Point3> factor1(poseKey, pointKey, measurement, model);
```
Comment (dellaert) [inline gtsam/sam/tests/testRangeFactor.cpp]: using Camera = PinholeCamera<Cal3_S2>;

File: cython/gtsam/examples/ImuFactorExample2.py (hunk 1 above)
Comment (varunagrawal) [reply cython/gtsam/examples/ImuFactorExample2.py]: Should I import all the classes? Keep the code consistent?

**Pull request merged**